src/consumerIncentives/saga.ts

```diff
@@ -48,15 +59,27 @@ export function* claimRewardsSaga({ payload: rewards }: ReturnType<typeof claimR
       [kit.web3.eth, kit.web3.eth.getTransactionCount],
       walletAddress
     )
+    const rewards: SuperchargePendingReward[] | SuperchargePendingRewardV2[] = yield select(
+      availableRewardsSelector
+    )
 
     Logger.debug(TAG, `Starting to claim ${rewards.length} rewards with baseNonce: ${baseNonce}`)
 
-    const receivedRewards: {
+    let receivedRewards: {
       fundsSource: string
       amount: string
       tokenAddress: string
       txHash: string
-    }[] = yield all(rewards.map((reward, index) => call(claimReward, reward, index, baseNonce)))
+    }[] = []
+
+    if (rewards.length > 0) {
+      const claimRewardFn = isSuperchargePendingRewardsV2(rewards) ? claimRewardV2 : claimReward
+      receivedRewards = yield all(
+        // @ts-ignore remove this ignore when upgrading to TS 4.2+ https://github.com/microsoft/TypeScript/issues/36390
```

Comment:
This way the compiler will tell us when we can remove it:
```suggestion
        // @ts-expect-error remove this when upgrading to TS 4.2+ https://github.com/microsoft/TypeScript/issues/36390
```

Reply:
thanks! i didn't know this, i updated